Objects/listobject.c

```diff
@@ -2649,6 +2675,15 @@ list___init___impl(PyListObject *self, PyObject *iterable)
         (void)_list_clear(self);
     }
     if (iterable != NULL) {
+        if(_PyObject_HasLen(iterable)){
+            Py_ssize_t iter_len = PyObject_Length(iterable);
```

Comment:
`PyObject_Length` is an alias of `PyObject_Size` (for backward compatibility). It is better to use `PyObject_Size`.